Sources/EdgeNetworkHandlers/CompletionHandlersManager.swift

```diff
@@ -48,13 +52,20 @@ class CompletionHandlersManager {
     func unregisterCompletionHandler(forRequestEventId: String) {
         guard !forRequestEventId.isEmpty else { return }
 
-        if let completionHandler = completionHandlers[forRequestEventId] {
-            completionHandler(edgeEventHandles[forRequestEventId] ?? [])
-            _ = completionHandlers.removeValue(forKey: forRequestEventId)
-            Log.trace(label: TAG, "Removing completion handler for Edge response with request event id \(forRequestEventId).")
+        queue.async { [weak self] in
+
+            guard let self = self else { return }
+
+            if let completionHandler = completionHandlers[forRequestEventId] {
+                completionHandler(edgeEventHandles[forRequestEventId] ?? [])
+
+                Log.trace(label: TAG, "Removing completion handler for Edge response with request event id \(forRequestEventId).")
+                completionHandlers.removeValue(forKey: forRequestEventId)
+            }
+
+            edgeEventHandles.removeValue(forKey: forRequestEventId)
```

Comment:
Same here, this may produce a lint warning as the result is ignored.

Reply:
RemoveValue has `@discardableResult` tag so this would not be an issue. 
<img width="722" alt="Screenshot 2024-06-10 at 11 30 47 AM" src="https://github.com/adobe/aepsdk-edge-ios/assets/4697559/a338b028-4eed-42e9-980e-035fb4acc208">